### ISSUE TITLE
Allow configuration of HttpClient by making HttpClientRestClient extensible

### ISSUE THREAD

**Issue by MosheElisha** [SourceLabOrg/kafka-connect-client#54]
Hi,

I would like to create a [TracingHttpClientBuilder](https://github.com/opentracing-contrib/java-apache-httpclient/blob/master/opentracing-apache-httpclient/src/main/java/io/opentracing/contrib/apache/http/client/TracingHttpClientBuilder.java) instead of the default `HttpClientBuilder.create();`.

One alternative is to create a protected method inside `HttpClientRestClient`:
```
@NotNull
protected HttpClientBuilder createHttpClientBuilder() {
     return HttpClientBuilder.create();
}
```

and use it inside the init method:
```
// Setup client builder
final HttpClientBuilder clientBuilder = createHttpClientBuilder();
```

With that method, I can create my own class that extends `HttpClientRestClient` and overrides the `createHttpClientBuilder` method.

If you prefer a different alternative or have a solution that does not require a PR, please let me know.

Thank you.

**Comment by MosheElisha** [comment]
@Crim can you please let me know if I can create a PR for the proposal above?

Thanks.

**Comment by Crim** [comment]
Hey there sorry for the late response I've been on holiday.

I've been trying to think of a better way to approach your problem, by using an injectable/configurable factory instance or something that doesn't involve you needing to extend and override an internal method of the library that I don't consider part of the public API.

But your approach is likely the easiest path forward.  You're welcome to submit a PR and I'll try to cut a release shortly for you with the change once merged.  I can't promise you that it won't ever break in future releases (being that I don't generally consider this to be part of the library's public api), but the reality is without any huge reason to change it, and things being pretty much feature complete, I wouldn't expect it to change out from under you.

Hope that makes sense,  thanks!

**Comment by MosheElisha** [comment]
No worries. Thanks for the reply, @Crim !

The BW compatible reservation you mentioned make sense and are fine by me. I will submit a PR in the next day or two.

**Comment by Crim** [comment]
Released 3.1.2 with this.  Thanks!
